### PR TITLE
Include xtrigger function signatures in `cylc config`

### LINKED ISSUE #5877: cylc config shows unhelpful information for xtriggers
### Description

If you have an xtrigger and run `cylc config my_workflow` the xtrigger is listed as `... = [xtrigger-func cmd]` which isn't very informative or helpful.

### Reproducible Example

I just used the test case from [the xtrigger plugin PR](https://github.com/cylc/cylc-flow/pull/5831).

Results was

```
$ cylc config cylctb-20231206T033820Z-uOHK/f/xtriggers/04-respect-cylc-pythonpath
[scheduler]
    cycle point format = %Y
[scheduling]
    initial cycle point = 2010
    final cycle point = 2011
    [[xtriggers]]
        x1 = [xtrigger-func cmd]
    [[graph]]
```

### Expected Behaviour

A resolved string of what the xtrigger was for reference after templating is resolved.

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #5877

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] No dependency changes 
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] No docs needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by wxtim — COMMENTED** [review]
- [x] Makes sense
- [x] Manually tested as working

No test?

File: cylc/flow/subprocctx.py [hunk 1]
```diff
@@ -158,3 +158,10 @@ def get_signature(self):
         args = self.func_args + [
             "%s=%s" % (i, self.func_kwargs[i]) for i in skeys]
         return "%s(%s)" % (self.func_name, ", ".join([str(a) for a in args]))
+
+    def __str__(self) -> str:
+        """
+        >>> str(SubFuncContext('label', 'func', [1, 2], {'a': 3}))
+        '[xtrigger-func cmd]: func(1, 2, a=3)'
+        """
+        return f"{SubProcContext.__str__(self)}: {self.get_signature()}"
```
Comment (hjoliver) [inline cylc/flow/subprocctx.py:167]: Do we need the `[xtrigger-func cmd]` prefix?
```suggestion
        return self.get_signature()
```

Comment (MetRonnie) [reply cylc/flow/subprocctx.py:167]: Dunno 😄

File: cylc/flow/subprocctx.py [hunk 2]
```diff
@@ -158,3 +158,17 @@ def get_signature(self):
...
+    def dump(self) -> str:
+        """Output for logging."""
+        return SubProcContext.__str__(self)
+
+    def __str__(self) -> str:
+        """
+        >>> str(SubFuncContext('label', 'my_func', [1, 2], {'a': 3}))
+        'my_func(1, 2, a=3):10.0'
+        """
+        return f"{self.get_signature()}:{self.intvl}"
```
Comment (MetRonnie) [inline cylc/flow/subprocctx.py:171]: The previous `__str__()` implementation was removed in https://github.com/cylc/cylc-flow/pull/5617/files#r1254330367, but I've ensured the output from xtriggers still gets logged in the workflow log by adding a `dump` method and using that instead

**Review by wxtim — APPROVED** [review]

File: cylc/flow/subprocctx.py (hunk 2 above)
Comment (oliver-sanders) [reply cylc/flow/subprocctx.py:171]: That was it! Great.

**Pull request merged**